File: Nabeatsu.java
import java.util.Scanner;

class Nabeatsu {

  private static boolean is_multiple(int num) {

    if (!(num % 3 == 0)) {
      return false;
    }
    return true;

  }

  private static boolean has_three(int num) {

    while (num > 0) {
      if (num % 10 == 3) {
        return true;
      }
      num /= 10;
    }

    return false;
  }

  private static void print_ismultiple(boolean ismultiple) {

    if (!ismultiple) {
      System.out.println("3の倍数ではありません");
    } else {
      System.out.println("3の倍数です");
    }

  }

  private static void print_has_three(boolean hasthree) {

    if (!hasthree) {
      System.out.println("3が含まれていません");
    } else {
      System.out.println("3が含まれています");
    }

  }

  public static void main(String[] args) {

    try (Scanner scanner = new Scanner(System.in)) {
      boolean roop = true;

      do {
        boolean is_multiple = false;
        boolean has_three = false;

        System.out.println("数字を入力してください");
        int input = scanner.nextInt();

        is_multiple = is_multiple(input);
        has_three = has_three(input);

        print_ismultiple(is_multiple);
        print_has_three(has_three);

        System.out.println("続けますか？ 1で継続、2で終了");
        int exit = scanner.nextInt();
        if (exit == 2) {
          roop = false;
        } else if (exit == 1)
          ;
      } while (roop);

    }

  }
}

File: Nabeatsu.c
#include <stdio.h>

int is_multiple(int num)
{
    if (num % 3 != 0)
    {
        return 0;
    }
    return 1;
}

int has_three(int num)
{
    while (num > 0)
    {
        if (num % 10 == 3)
        {
            return 1;
        }
        num /= 10;
    }
    return 0;
}

void print_ismultiple(int ismultiple)
{
    if (ismultiple == 0)
    {
        printf("3の倍数ではありません\n");
    }
    else
    {
        printf("3の倍数です\n");
    }
}

void print_has_three(int hasthree)
{
    if (hasthree == 0)
    {
        printf("3が含まれていません\n");
    }
    else
    {
        printf("3が含まれています\n");
    }
}

int main()
{
    int roop = 1;

    do
    {
        int is_multiple_val = 0;
        int has_three_val = 0;

        printf("数字を入力してください：");
        int input;
        scanf("%d", &input);

        is_multiple_val = is_multiple(input);
        has_three_val = has_three(input);

        print_ismultiple(is_multiple_val);
        print_has_three(has_three_val);

        printf("続けますか？ 1で継続、2で終了：");
        int exit;
        scanf("%d", &exit);

        if (exit == 2)
        {
            roop = 0;
        }
    } while (roop);

    return 0;
}
